### DISCUSSION TITLE
Why does SqsContainerOptions.pollTimeout have a max value of 10s, when according to AWS docs this can be up to 20s?

### DISCUSSION THREAD

**Discussion by bmcswee-bill** [awspring/spring-cloud-aws#1279] [Q&A]
According to the Spring Cloud AWS docs [here](https://docs.awspring.io/spring-cloud-aws/docs/3.2.0/reference/html/index.html#sqscontaineroptions-descriptions), `pollTimeout` has a max value of 10s.

According to the AWS docs [here](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/best-practices-setting-up-long-polling.html), polling can be configured to wait for up to 20s.

Why the difference? Also, it appears the Spring Cloud AWS does not actually enforce a maximum value on the config option, and there's no mention of a maximum in the Javadocs - only in the HTML docs linked. This leaves me wondering if this is a mistake in the docs.

**Comment by shreyas957** [comment]
@bmcswee-bill  If we did not specify any value then the default of 10 sec is used. But at the same time they have not enforced any polling time limit. And I think they need to enforce it in future. 
![image](https://github.com/user-attachments/assets/a895fc5d-8fa0-41bf-a9f0-16b724ec5525)

**Comment by tomazfernandes** [comment]
Hey @bmcswee-bill and @shreyas957, yup, that's a mistake.

Can you please open a couple of issues for fixing the docs and for enforcing the limits?

PRs are welcome!

**Reply by shreyas957** [reply]
Sure I'll do that
